explorer/client/src/reducers/adminHeads.ts

```diff
@@ -3,20 +3,41 @@ import { Reducer } from 'redux'
 import { Head } from 'explorer/models'
 
 export interface State {
-  items?: Record<string, Head>
+  loading: boolean
+  error: boolean
+  items?: Record<Head['id'], Head>
 }
 
-const INITIAL_STATE: State = {}
+const INITIAL_STATE: State = {
+  loading: false,
+  error: false,
+}
 
 export const adminHeads: Reducer<State, Actions> = (
   state = INITIAL_STATE,
   action,
 ) => {
   switch (action.type) {
+    case 'FETCH_ADMIN_HEADS_BEGIN':
+      return {
+        ...state,
+        loading: true,
+      }
+
     case 'FETCH_ADMIN_HEADS_SUCCEEDED':
-      return { items: { ...action.data.heads } }
-    case 'FETCH_ADMIN_HEAD_SUCCEEDED':
-      return { items: { ...action.data.heads } }
+      return {
+        ...state,
+        items: { ...action.data.heads },
+        loading: false,
+      }
+
+    case 'FETCH_ADMIN_HEADS_ERROR':
+      return {
+        ...state,
+        loading: false,
+        error: true,
```

Comment:
What do you think about using error object here? `error: action.error`

Reply:
It's recommended to only use serializable state in the store https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state